runtime/interpreter/interpreter.go

```diff
@@ -3271,8 +3276,31 @@ func (interpreter *Interpreter) newStorageIterationFunction(addressValue Address
 			}()
 
 			for key, value := storageIterator.Next(); key != "" && value != nil; key, value = storageIterator.Next() {
+				staticType := value.StaticType(inter)
+
+				// Perform a forced type loading to see if the underlying type is not broken.
+				// If broken, skip this value from the iteration.
+				var typeError error
+				func() {
+					defer func() {
+						if r := recover(); r != nil {
+							switch r := r.(type) {
+							case errors.UserError, errors.ExternalError:
+								typeError = r.(error)
+							default:
+								panic(r)
+							}
+						}
+					}()
+					_, typeError = inter.ConvertStaticToSemaType(staticType)
+				}()
```

Comment:
is it necessary to have this double nested function here? What errors are we throwing inside `ConvertStaticToSemaType` that aren't being returned in the `err` value?

Reply:
Yup, the nested function is needed to recover errors thrown from  `ConvertStaticToSemaType`. For example, If the import handler fails to load the program (contract), it panics:
https://github.com/onflow/cadence/blob/4ab56fd102820c614d48748dbcfedd9d933344ed/runtime/environment.go#L807-L815

The two tests added cover these two cases; one where the error is returned, and the other for the case where the error is thrown.

Reply:
Extracted the type loading function out, so its no longer a lambda: c236cd4